actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -110,15 +112,36 @@ def non_recursive(cache, options)
           routes
         end
 
+        module RegexCaseComparator
+          DEFAULT_INPUT = /[-_.a-zA-Z0-9]+\/[-_.a-zA-Z0-9]+/
+          DEFAULT_REGEX = /\A#{DEFAULT_INPUT}\Z/
+
+          def self.===(regex)
+            DEFAULT_INPUT == regex
+          end
+        end
+
         # Returns an array populated with missing keys if any are present.
         def missing_keys(route, parts)
-          missing_keys = []
+          missing_keys = nil
           tests = route.path.requirements
           route.required_parts.each { |key|
-            if tests.key?(key)
-              missing_keys << key unless /\A#{tests[key]}\Z/ === parts[key]
+            case tests[key]
+            when nil
```

Comment:
I don't think you can have a required key be nil. I ran this against the test suite and came up with nothing:

```ruby
          puts tests.inspect if tests.values.include?(nil)
```